proxy/upgrade.go

```diff
@@ -132,6 +132,13 @@ func (p *upgradeProxy) serveHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	// Backend sent Connection: close
+	if resp.Close {
+		w.WriteHeader(http.StatusServiceUnavailable)
+		w.Write([]byte(http.StatusText(http.StatusServiceUnavailable)))
+		return
+	}
```

Comment:
According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/101
the correct `Connection` header value is `Upgrade`:
```
HTTP/1.1 101 Switching Protocols
Upgrade: websocket
Connection: Upgrade
```